Core/Net/Net.cs

```diff
@@ -194,11 +194,11 @@ public static void DownloadWithProgress(ICollection<DownloadTarget> downloadTarg
             }.DownloadAndWait(downloadTargets);
         }
 
-        public static string DownloadText(Uri url, string authToken = "", string mimeType = null)
+        public static string DownloadText(Uri url, string authToken = "", string mimeType = null, int timeout = Timeout.Infinite)
```

Comment:
Let's document the units of the new param (milliseconds) in a `/// <param>` comment.

Reply:
I think the current default is 100 seconds, not infinite:

https://docs.microsoft.com/en-us/dotnet/api/system.net.webrequest.timeout?view=netframework-4.7.2

> The default value is defined by the descendant class.

https://docs.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest.timeout?view=netframework-4.7.2#System_Net_HttpWebRequest_Timeout

> The default value is 100,000 milliseconds (100 seconds).

Not sure how to reconcile that with multi-minute delays in indexing. Retries, maybe? 

But we should consider whether changing it from 100 seconds to infinite is desirable.

Reply:
This sounds suspicious, I wonder why they felt the need to spell this out:

> A Domain Name System (DNS) query may take up to 15 seconds to return or time out. If your request contains a host name that requires resolution and you set Timeout to a value less than 15 seconds, it may take 15 seconds or more before a WebException is thrown to indicate a timeout on your request.

Reply:
> I think the current default is 100 seconds, not infinite

Huh, I was sure I've read Infinite. Anyway, that gets us closer to the 7 minutes we see with the bot, 3*100s = 5m

> But we should consider whether changing it from 100 seconds to infinite is desirable.

Yeah I think I'm going to change it to a default of 100s then.

> This sounds suspicious, I wonder why they felt the need to spell this out

I think it means that the timeout is not applied to the DNS resolution. So if it's that one that fails / times out, the request only fails after 15 seconds, not already 10 seconds.
If you want, we can also set the timeout to 15 seconds, but I don't think it's necessary. If It's not DNS that fails, we can still get the 10s timeout.